docs/spec/consensus/checkvalidators.md

```diff
@@ -0,0 +1,224 @@
+# Checking Validator Set (CheckVS)
+
+As part of the light client, the CheckVS procedure has to check, whether given
+two headers (whose heigts differ by more than 1), the LightClient can trust  the
+newer header under the assumption it trusted the old one.
+
+This document contains some math formulas. To ease reading, the file
+  /tendermint/docs/spec/pdfs/checkvalidators.pdf
+displays them correctly.
+
+## Definitions
+
+  * header fields
+    - $height$
+    - $bfttime$: the chain time when the header (block) was generated
+    - $V$: validator set containing validators.
+    - $nextV$: next validators
+  * $tp$: trusting period   
+  * for a time $t$, the predicate $correct(v,t)$ is true if the validator $v$
+    follows the protocol until time $t$ (we will see about recovery later).
+    Similarly we define $faulty(v,t)$.
+  * For each header $h$ it has locally stored, the LightClient stores whether
+    it trusts $h$. We write $trust(h) = true$, if this is the case.
+  * Validator fields. We will write a validator as a tuple $(v,p)$ such that
+    + $v$ is the identifier (we assume identifiers are unique in each validator set)
+    + $p$ is its voting power
+
+## LightClient Trusting Spec
+
+### LightClient Invariant
+ For each LightClient $l$ and each header $h$:
+if $l$ has set $trust(h) = true$,
+  then validators that are correct until time $h.bfttime + tp$ have more than two thirds of the voting power in $h.V$. (Or/and $h.nextV$)
+
+  Formally,
+  \[\sum_{(v,p) \in h.V \wedge \\correct(v,h.bfttime + tp)} p > 2/3 \sum_{(v,p) \in h.V} p\]
+
+  Equivalently,
+  \[\sum_{(v,p) \in h.V \wedge\\ faulty(v,h.bfttime + tp)} p < 1/3 \sum_{(v,p) \in h.V} p\]
+
+**Question:** What should be the precise assumption here. Is the invariant on $h.V$ or on $h.NextV$ or both?
+
+*Assumption:* If a header is properly generated, then the above equations hold.
+
+### Liveness
+
+*Draft:* If a header $h$ as been properly generated by the blockchain (and its age is less than the trusting period), then a correct LightClient will eventually set $trust(h) = true$.
```

Comment:
We need to assume synchronous data availability of headers, which is fine. Maybe we should emphasize that headers can be received out of order and the light client will determine which it needs to query for at what times?